soc/riscv/riscv-privilege/hi3861/soc.c

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2023 Chen Xingyu <hi@xingrz.me>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/arch/common/sys_io.h>
+
+#define GLB_CTL_BASE                (0x50000000)
+#define GLB_CTL_GLB_WDT_RST_SEL_REG (GLB_CTL_BASE + 0x30)
```

Comment:
hardcoding an address and directly accessing that is really something that we should avoid when possible. What peripheral does this address belong to? What's GLB?

Reply:
alright, maybe you could use a syscon device for this block?

Reply:
Great idea, I will update it later. Thanks!